arkouda/numeric.py

```diff
@@ -425,6 +438,418 @@ def cos(pda: pdarray) -> pdarray:
+
+    Parameters
+    ----------
+    pda : pdarray
+
+    Returns
+    -------
+    pdarray
+        A pdarray containing inverse tangent for each element
+        of the original pdarray
+
+    Raises
+    ------
+    TypeError
+        Raised if the parameter is not a pdarray
+    """
+    repMsg = generic_msg(
+        cmd="efunc",
+        args={
+            "func": "arctan",
+            "array": pda,
+        },
+    )
+    return create_pdarray(type_cast(str, repMsg))
+
+
+@typechecked
+def arctan2(num: Union[numeric_scalars, pdarray], denom: Union[numeric_scalars, pdarray]) -> pdarray:
+    """
+    Return the element-wise inverse tangent of the array pair. The result chosen is the
+    signed angle in radians between the ray ending at the origin and passing through the
+    point (1,0), and the ray ending at the origin and passing through the point (denom, num).
+    The result is between -pi and pi.
+
+    Parameters
+    ----------
+    num : Union[numeric_scalars, pdarray]
+        Numerator of the arctan2 argument.
+    denom : Union[numeric_scalars, pdarray]
+        Denominator of the arctan2 argument.
+    Returns
+    -------
+    pdarray
+        A pdarray containing inverse tangent for each corresponding element pair
+        of the original pdarray, using the signed values or the numerator and
+        denominator to get proper placement on unit circle.
+
+    Raises
+    ------
+    TypeError
+        Raised if the parameter is not a pdarray
+    """
+    if (isinstance(num, pdarray) or isSupportedNumber(num)) and (
+        isinstance(denom, pdarray) or isSupportedNumber(denom)
+    ):
+        if isinstance(num, pdarray) and isinstance(denom, pdarray):
+            repMsg = generic_msg(
+                cmd="efunc2vv",
+                args={
+                    "func": "arctan2",
+                    "num": num,
+                    "denom": denom,
+                },
+            )
+        if isinstance(num, pdarray) and isSupportedNumber(denom):
+            repMsg = generic_msg(
+                cmd="efunc2vs",
+                args={
+                    "func": "arctan2",
+                    "num": num,
+                    "scalar": denom,
+                    "dtype": resolve_scalar_dtype(denom),
+                },
+            )
+        if isSupportedNumber(num) and isinstance(denom, pdarray):
+            repMsg = generic_msg(
+                cmd="efunc2sv",
+                args={
+                    "func": "arctan2",
+                    "scalar": num,
+                    "denom": denom,
+                    "dtype": resolve_scalar_dtype(num),
+                },
+            )
+        return create_pdarray(type_cast(str, repMsg))
+    else:
+        raise TypeError(
+            f"Unsupported types {type(num)} and/or {type(denom)}. Supported "
+            f"types are numeric scalars and pdarrays"
+        )
```

Comment:
The logic here is not completely correct. The outer `if` will allow the for the cases covered inside it, but it will also allow for `isSupportedNumber(num) and isSupportedNumber(num)`. I do not believe that we want to include that case here, but we need to logic to properly error. The current workflow would try to run `create_pdarray`, but the `repMsg` variable does not exist. 

Here is an example for a single block which should be useable for each case which would be in an `if..elif..else` structure.

```python
if isinstance(num, pdarray) and isinstance(denom, pdarray):
    return create_pdarray(type_cast(str, generic_msg(
                cmd="efunc2vv",
                args={
                    "func": "arctan2",
                    "num": num,
                    "denom": denom,
                },
            )))
```